x-pack/plugins/cross_cluster_replication/public/app/app.js

```diff
@@ -19,47 +36,186 @@ import {
-          <Route exact path={`${BASE_PATH}/follower_indices/add`} component={FollowerIndexAdd} />
-          <Route exact path={`${BASE_PATH}/follower_indices/edit/:id`} component={FollowerIndexEdit} />
-          <Route exact path={`${BASE_PATH}/:section`} component={CrossClusterReplicationHome} />
-        </Switch>
-      </div>
-    );
-  }
-}
+    componentWillUnmount() {
+      routing.userHasLeftApp = true;
+    }
+
+    async checkPermissions() {
+      this.setState({
+        isFetchingPermissions: true,
+      });
+
+      try {
+        const { hasPermission, missingPermissions } = await loadPermissions();
 
+        this.setState({
+          isFetchingPermissions: false,
+          hasPermission,
+          missingPermissions,
+        });
+      } catch (error) {
+        // Expect an error in the shape provided by Angular's $http service.
+        if (error && error.data) {
+          return this.setState({
+            isFetchingPermissions: false,
+            fetchPermissionError: error.data,
+          });
+        }
+
+        // This error isn't an HTTP error, so let the fatal error screen tell the user something
+        // unexpected happened.
+        fatalError(error, i18n.translate('xpack.crossClusterReplication.app.checkPermissionsFatalErrorTitle', {
+          defaultMessage: 'Cross Cluster Replication app',
+        }));
+      }
+    }
+
+    registerRouter() {
+      const { router } = this.context;
+      routing.reactRouter = router;
+    }
+
+    render() {
+      const {
+        isFetchingPermissions,
+        fetchPermissionError,
+        hasPermission,
+        missingPermissions,
+      } = this.state;
+
+      if (isFetchingPermissions) {
+        return (
+          <EuiPageContent horizontalPosition="center">
+            <EuiFlexGroup>
+              <EuiFlexItem>
+                <EuiLoadingSpinner size="xl"/>
+              </EuiFlexItem>
+
+              <EuiFlexItem>
+                <EuiTitle>
+                  <h2>
+                    <FormattedMessage
+                      id="xpack.crossClusterReplication.app.permissionCheckTitle"
+                      defaultMessage="Checking permissions..."
+                    />
+                  </h2>
+                </EuiTitle>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiPageContent>
+        );
+      }
+
+      if (fetchPermissionError) {
+        const { error: errorString, statusCode, message } = fetchPermissionError;
+
+        return (
+          <EuiCallOut
+            title={(
+              <FormattedMessage
+                id="xpack.crossClusterReplication.app.permissionCheckErrorTitle"
+                defaultMessage="Error checking permissions"
+              />
+            )}
+            color="danger"
+            iconType="alert"
+          >
+            {`${statusCode}: ${errorString}. ${message}`}
```

Comment:
it would also be great to traverse through `cause` in the error object and surface that:

![image](https://user-images.githubusercontent.com/1965714/51768612-a4e1fb00-2095-11e9-92f7-8dc26cfde64f.png)

Reply:
Done!

![image](https://user-images.githubusercontent.com/1238659/51778722-25195800-20b8-11e9-93f9-90a246b77d96.png)